csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -52,6 +52,65 @@ internal PropertyValue(string v, bool u)
         public bool? GetPropertyAsBool(string name) =>
             GetPropertyAsInt(name) is int intValue ? intValue > 0 : (bool?)null;
 
+        public int? GetPropertyAsByteSize(string name)
+        {
+            lock (_properties)
+            {
+                if (_properties.TryGetValue(name, out PropertyValue? pv))
+                {
+                    pv.Used = true;
+
+                    if (pv.Val == "unlimited")
+                    {
+                        return 0;
+                    }
+
+                    // Match an integer followed letters
+                    Match match = Regex.Match(pv.Val, @"^([0-9]+[\.]?[0-9]*)([A-Z]+)$");
```

Comment:
I would prefer a simpler format/parsing for the byte-size properties, with a single-letter optional suffix and no "unlimited" option (and no need to use regular expressions):
- integer without suffix == number of bytes
- integer with "K", "M", "G" suffix == number of KB/MB/GB bytes
(K, M, G are common byte-size suffixes, used for example by https://linux.die.net/man/8/resize2fs)

for example:
```
Ice.MessageSizeMax=16K # 16 KB
```

And the number must be a positive integer, for example:
```
Ice.MessageSizeMax=10.0M # error
```

Reply:
Ice.MessageSizeMax in 3.7 gets KB , and this setting will be now interpreted as Bytes, other than that I'm good with simplifying this as you suggested

Reply:
What about `unlimited` should we keep this special value or accept `0` `0K` ...

Reply:
I think `0` maybe is fine.